openedx/core/djangoapps/lang_pref/api.py

```diff
@@ -2,17 +2,41 @@
 """ Python API for language and translation management. """
 
 from collections import namedtuple
+import logging
 
 from django.conf import settings
 from django.utils.translation import ugettext as _
+
 from openedx.core.djangoapps.dark_lang.models import DarkLangConfig
+from openedx.core.djangoapps.site_configuration.helpers import get_value
 
 
 # Named tuples can be referenced using object-like variable
 # deferencing, making the use of tuples more readable by
 # eliminating the need to see the context of the tuple packing.
 Language = namedtuple('Language', 'code name')
 
+log = logging.getLogger(__name__)
+
+
+def header_language_selector_is_enabled():
+    """Return true if the header language selector has been enabled via settings or site-specific configuration."""
+    setting = get_value('SHOW_HEADER_LANGUAGE_SELECTOR', settings.FEATURES.get('SHOW_HEADER_LANGUAGE_SELECTOR', False))
+
+    # The SHOW_LANGUAGE_SELECTOR setting is deprecated, but might still be in use on some installations.
+    deprecated_setting = get_value('SHOW_LANGUAGE_SELECTOR', settings.FEATURES.get('SHOW_LANGUAGE_SELECTOR', None))
+    if deprecated_setting is None:
+        deprecated_setting = False
+    else:
+        log.warn("The SHOW_LANGUAGE_SELECTOR setting is deprecated. Use SHOW_HEADER_LANGUAGE_SELECTOR instead.")
```

Comment:
@jibsheet Should we be concerned about this spamming the log for users that haven't updated their settings? I believe this line would execute on every page since it is part of the header.